### ISSUE TITLE
[Bug]: modal file-open dialog does not receive keyboard focus on Linux

### ISSUE THREAD

**Issue by gtritchie** [electron/electron#42948]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

30.2.0, 31.2.1

### What operating system(s) are you using?

Ubuntu

### Operating System Version

Ubuntu 22.04.4 LTS

### What arch are you using?

arm64 (including Apple Silicon)

### Last Known Working Electron version

30.0.3

### Expected Behavior

A modal file-open dialog (dialog.showOpenDialog with a parent BrowserWindow specified) receives keyboard focus when displayed.

### Actual Behavior

Modal file-open dialog does not receive keyboard focus and must be clicked on to operate via the keyboard. This is irritating for keyboard-preferring users and a barrier for keyboard-only users.

In the gist, click the "Open..." button with the mouse or keyboard, then try using the resulting file picker with the keyboard. For example, try hitting ESC to close it, or Tab to move around the dialog. Nothing happens.

### Testcase Gist URL

https://gist.github.com/afde5fa1a0a81b503b409105da3062d5

### Additional Information

If you Alt+Tab away from the application, then back to it (while the modal dialog is displayed), then the dialog gets focus and can be operated. So there's a workaround, but it's not ideal.

**Comment by gtritchie** [comment]
This is now working in v31.6.0 and newer, v32.2.1 and newer, and 33.0.0 and newer.
